engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/SelectModulesScreen.java

```diff
@@ -513,6 +512,7 @@ private void updateModuleInformation() {
                     int pos = Collections.binarySearch(sortedModules, info, moduleInfoComparator);
                     if (pos < 0) {                             // not yet in the (sorted) list
                         sortedModules.add(-pos - 1, info);     // use "insertion point" to keep the list sorted
+                        allSortedModules = new ArrayList<>(sortedModules);
```

Comment:
You're creating a new `ArrayList` for every module here - it should be outside the loop, I guess.
Using `clear()` and `addAll()` would avoid some memory allocation, so I would prefer that.